src/elements/Progressbar/index.jsx

```diff
@@ -0,0 +1,46 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import cx from 'classnames/bind';
+import style from './style.scss';
+import numberInRange from '../../utils/numberInRange';
+
+const classNames = cx.bind(style);
+const colorMap = {
+  success: 'color-success',
+  warning: 'color-warning',
+  error: 'color-error',
+};
+
+const Progressbar = ({ percentage = 0, showPercentageNum, className, color, ...props }) => {
+  const progressBarWidth = percentage > 100 ? 100 : percentage;
+  const percentageLabel = showPercentageNum ? `${percentage}%` : null;
+  const progressBarColor = colorMap[color] || 'color-success';
+  const dynamicStyle = {
+    width: `${progressBarWidth}%`,
+  };
```

Comment:
we also don't need this variable. we can directly do it in jsx as you did before.